src/middlewared/middlewared/plugins/vm/vm_devices.py

```diff
@@ -48,6 +48,19 @@ class Config:
         datastore_extend = 'vm.device.extend_device'
         cli_namespace = 'service.vm.device'
 
+    @accepts()
+    @returns(Dict())
+    async def disk_choices(self):
+        """
+        Returns disk choices for device type "DISK".
+        """
+        return {
+            vol['id']: os.path.join('/dev/zvol', vol['id'].replace(' ', '+'))
+            for vol in await self.middleware.call(
+                'pool.dataset.query', [['type', '=', 'VOLUME']], {'extra': {'properties': []}}
```

Comment:
Let's please also check if the volume is locked or not as we shouldn't be showing locked volume as a valid choice